docs/reference/issuers/acme/dns01/azuredns.rst

```diff
@@ -0,0 +1,70 @@
+
+For security purposes, it is appropriate to utilize RBAC to ensure that you properly maintain access control to your resources in Azure. The service principal that is generated by this tutorial has fine grained access to ONLY the DNS Zone in the specific resource group specified. It requires this permission so that it can read/write the _acme_challenge TXT records to the zone.
+
+To create the service principal:
+
+.. code-block:: bash
+  :linenos:
+
+  AZURE_CERT_MANAGER_SP_NAME=SOME_SERVICE_PRINCIPAL_NAME
+  AZURE_CERT_MANAGER_SP_PASSWORD=SOME_PASSWORD
+  AZURE_CERT_MANAGER_DNS_RESOURCE_GROUP=SOME_RESOURCE_GROUP
+  AZURE_CERT_MANAGER_DNS_NAME=SOME_DNS_ZONE
+
+  AZURE_CERT_MANAGER_SP_APP_ID=$(az ad sp create-for-rbac --name $AZURE_CERT_MANAGER_SP_NAME --password $AZURE_CERT_MANAGER_SP_PASSWORD --query "appId" --output tsv)
+
+  # Lower the Permissions of the SP
+  az role assignment delete --assignee $AZURE_CERT_MANAGER_SP_APP_ID --role Contributor
+
+  # Give Access to DNS Zone
+  DNS_ID=$(az network dns zone show --name $AZURE_CERT_MANAGER_DNS_NAME --resource-group $AZURE_CERT_MANAGER_DNS_RESOURCE_GROUP --query "id" --output tsv)
+
+  az role assignment create --assignee $AZURE_CERT_MANAGER_SP_APP_ID --role "DNS Zone Contributor" --scope $DNS_ID
+
+  # Check Permissions
+  az role assignment list --assignee $AZURE_CERT_MANAGER_SP_APP_ID
+
+  # Create Secret
+  kubectl create secret generic azuredns-config \
+    --from-literal=CLIENT_SECRET=$AZURE_CERT_MANAGER_SP_PASSWORD
+
+  # Get the Service Principal App ID for configuration
+  echo $AZURE_CERT_MANAGER_SP_APP_ID
+
+You can configure the issuer like so:
+
+.. code-block:: yaml
+
+  apiVersion: certmanager.k8s.io/v1alpha1
+  kind: ClusterIssuer
+  metadata:
+    name: letsencrypt-prod
+  spec:
+    acme:
+      server: https://acme-v02.api.letsencrypt.org/directory
+      email: example@example.com 
+      privateKeySecretRef:
+        name: letsencrypt-prod
+      dns01:
+          providers:
+              - name: azure
+                azuredns:
+                  # Service principal clientId (also called appId)
+                  clientID: AZURE_SERVICE_PRINCIPAL_ID
+                  # A secretKeyRef to a service principal ClientSecret (password)
+                  # ref: https://docs.microsoft.com/en-us/azure/container-service/kubernetes/container-service-kubernetes-service-principal
+                  clientSecretSecretRef:
+                    name: AZUREDNS_SECRET_KEY_NAME
+                    key: CLIENT_SECRET
+                  # Azure subscription Id
+                  subscriptionID: AZURE_SUBSCRIPTION_ID
+                  # Azure AD tenant Id
+                  tenantID: AZURE_TENANT_ID
+                  # ResourceGroup name where dns zone is provisioned
+                  resourceGroupName: AZURE_RESOURCE_GROUP
+                  hostedZoneName: AZURE_DNS_ZONE_NAME
```

Comment:
This field is optional, so it'd be good to call this out

Reply:
@munnerz What does that field do exactly? I'm not super familiar with Go and in my Azure implementations I just default to using it.

Reply:
So hostedZoneID shouldn't really exist as an option on the issuer IMO. It was originally included as part of the Route53 DNS provider only, and it is meant to be used in the cases where there are multiple zones configured for the same domain, e.g. on Route53 some people configure an internal and external DNS zone for the same domain (e.g. split horizon DNS).

If this field is not included, cert-manager will perform a List operation on the account in order to 'look up' the correct hosted zone name to use for the domain. If more than one matches, iirc, the 'first' one is chosen.